vulcanai2/tests/models/test_layers.py

```diff
@@ -1,42 +1,196 @@
 import pytest
 import numpy as np
+import math
+import copy
 import torch
 from vulcanai2.models.layers import *
+from vulcanai2.models.utils import selu_weight_init_, selu_bias_init_
+from vulcanai2.models.dnn import DenseNet
+from vulcanai2.models.cnn import ConvNet
+from torch.utils.data import TensorDataset, DataLoader
+
 
 class TestBaseUnit:
+    """To test BaseUnit layer."""
+
     @pytest.fixture
-    def baseunit(self): 
+    def baseunit(self):
+        """Base unit fixture."""
         return BaseUnit()
 
+
 class TestDenseUnit:
+    """To test DenseUnit layer."""
+
     @pytest.fixture
     def dense_unit(self):
+        """Dense unit fixture."""
         return DenseUnit(
             in_features=10,
             out_features=10
         )
-    
+
     def test_forward(self, dense_unit):
         """Confirm size is expected after forward."""
         test_input = torch.ones([1, dense_unit.in_features])
         output = dense_unit.forward(test_input)
         assert output.size() == torch.ones([1, dense_unit.out_features]).size()
-    
+
 
 class TestConvUnit:
+    """To test ConvUnit layers."""
+
     @pytest.fixture
     def conv_unit(self):
+        """Create ConvUnit fixture."""
         return ConvUnit(
             conv_dim=2,
             in_channels=10,
             out_channels=10,
-            kernel_size=(5,5)
+            kernel_size=(5, 5)
         )
-    
+
     def test_forward(self, conv_unit):
         """Confirm size is expected after forward."""
         test_input = torch.ones([1, conv_unit.in_channels, 28, 28])
         output = conv_unit.forward(test_input)
         # No padding with 2 5x5 kernels leads from 28x28 -> 24x24
         assert output.size() == torch.ones([1, conv_unit.out_channels, 24, 24]).size()
 
+
+class TestSeluInit:
+    """To test selu initialized layer properties."""
+
+    @pytest.fixture
+    def dense_unit(self):
+        """Create a dense unit fixture."""
+        return DenseUnit(
+            in_features=10,
+            out_features=10
+        )
+
+    @pytest.fixture
+    def conv_unit(self):
+        """Create a conv unit fixture."""
+        return ConvUnit(
+            conv_dim=2,
+            in_channels=10,
+            out_channels=10,
+            kernel_size=(5, 5)
+        )
+
+    def test_dense_selu_weight_change(self, dense_unit):
+        """Confirm SELU weight init properties hold for dense net."""
+        starting_weight = copy.deepcopy(dense_unit._kernel.weight)
+        std = round(math.sqrt(1. / 10), 1)
+
+        dense_unit.weight_init = selu_weight_init_
+        dense_unit._init_weights()
+        new_weight = copy.deepcopy(dense_unit._kernel.weight)
+        assert (torch.equal(starting_weight, new_weight) is False)
+        assert pytest.approx(round(new_weight.std().item(), 1), 0.1) == std
```

Comment:
What does calling `.item()` give you?

Reply:
@rfratila `new_weight.std()` is a tensor so `.item()` gives you the value itself.